dev-tools/xtask/src/a4x2_deploy.rs

```
@@ -0,0 +1,472 @@
+    package: Utf8PathBuf,
+}
+
+struct Environment {
+    a4x2_package_tar: Utf8PathBuf,
+    work_dir: Utf8PathBuf,
+    a4x2_dir: Utf8PathBuf,
+    out_dir: Utf8PathBuf,
+    in_ci: bool,
+}
+
+pub fn run_cmd(args: A4x2DeployArgs) -> Result<()> {
+    let sh = Shell::new()?;
+
+    // TODO: do we want to have a configurable falcon dataset?
+
+    let env = {
+        // Buildomat sets the CI environment variable. We'll work a bit
+        // differently in that case (no cleanup, use /work)
+        let in_ci = env::var("CI").is_ok();
+
+        let a4x2_package_tar = match &args.command {
+            // Unused by stop command, so harmless to fill in with a default
+            DeployCommand::Stop => Utf8PathBuf::from(DEFAULT_A4X2_PKG_PATH),
+
+            DeployCommand::Start(args) => args.package.clone(),
+            DeployCommand::RunTests(args) => args.package.clone(),
+        };
+        let a4x2_package_tar = a4x2_package_tar.canonicalize_utf8()?;
+
+        let home_dir = Utf8PathBuf::from(env::var("HOME")?);
+        let work_dir = if in_ci {
+            Utf8PathBuf::from("/work/a4x2-deploy")
+        } else {
+            home_dir.join(".cache/a4x2-deploy")
+        };
+
+        // a4x2 dir. will be created by the unpack step
+        let a4x2_dir = work_dir.join("a4x2-package-out");
+
+        // Output. Maybe in CI we want this to be /out
+        let out_dir = work_dir.join("a4x2-deploy-out");
+
+        Environment { a4x2_package_tar, work_dir, a4x2_dir, out_dir, in_ci }
+    };
+
+    match args.command {
+        DeployCommand::Stop => teardown_a4x2(&sh, &env)?,
+        DeployCommand::Start(_) | DeployCommand::RunTests(_) => {
+            if !env.in_ci {
+                // Teardown previous deploy if it exists, before wiping the data
+                // for it. If this errors, we assume the deploy doesn't exist
+                // and carry on.
+                let result = teardown_a4x2(&sh, &env);
```

I don't think we should ignore this error.  Either **teardown_a4x2()** should work, or we need to stop and figure out why it did not.